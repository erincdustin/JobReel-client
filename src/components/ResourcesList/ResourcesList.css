.resources-list {
  position: relative;
  grid-column-start: 4;
  grid-column-end: 9;
  height: 500px;
  padding: 10px;
  border-radius: 10px;
  position: relative;
  text-align: center;
  overflow-y: auto;
}

.savedResourceFilterControls {
  width: 90%;
  text-align: right;
}

@media screen and (max-width: 960px) {
  .resources-list {
    grid-column-start: 1;
    grid-column-end: 9;
  }
}
