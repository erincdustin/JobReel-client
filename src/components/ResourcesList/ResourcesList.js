import React from 'react';
import './ResourcesList.css';
import ResourceCard from '../ResourceCard/ResourceCard';
import Button from '../Button/Button';
import ResourcesForm from '../ResourcesForm/ResourcesForm';


export default class ResourcesList extends React.Component {
  state = {
    error: null,
  }


  
  render() {
    return (
      <div className='resources-list'>
        <h3>Resources</h3>
        <div className='button-for-add-form'>
          <Button type='button' onClick={this.unfoldForm}>Add Resource</Button>
        </div>
        <div className='saved-resources'>
          <ResourceCard/>
        </div>
      </div>
    )
  }
}