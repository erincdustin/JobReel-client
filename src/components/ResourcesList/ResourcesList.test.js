import React from 'react'
import ReactDOM from 'react-dom'
import ResourcesList from './ResourcesList'
import PropTypes from 'prop-types'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

class ContextProvider extends React.Component{
  static childContextTypes = {
    myContext: PropTypes.object,
  }
  
  getChildContext = () => ({
    myContext: {
      resources: [
        {
          title: 'title',
          type: 'type',
          description: 'description',
          resource_id: 1,
          date_added: '2019-08-26T15:03:05.646Z',
          user_id: 1
        }
      ],
      setManualResourceAdd: () => { }
    }
  })
  
  render(){
    return this.props.children
    }
  }

describe('ResourcesList smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<ResourcesList />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe('ResourcesList snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(<ContextProvider><ResourcesList /></ContextProvider>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})