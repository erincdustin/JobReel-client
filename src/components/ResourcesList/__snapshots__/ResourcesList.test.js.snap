// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourcesList snapshot test renders the UI as expected 1`] = `<ResourcesList />`;
